node_tree.py

```diff
@@ -299,15 +356,19 @@ def update(self):
 
         # this is a no-op if there's no drawing
         clear_exception_drawing_with_bgl(self.nodes)
-
+        if get_first_run():
+            return
         if self.skip_tree_update:
             # print('throttled update from context manager')
             return
 
-        # print('svtree update', self.timestamp)
-        self.has_changed = True
-        # self.has_link_count_changed
-        self.process()
+        if link_cache_is_ready(self):
+            use_link_memory(self)
+        else:
+            if self.sv_process:
+                fill_links_memory(self)
+            self.has_changed = True
+            self.process()
```

Comment:
What I'm expecting to see here:
```Python
update_sv_links()  # probably
if links_has_changed():
    process_from_nodes(get_effected_nodes())
else:
    self.has_changed = True
    self.process()
```

Reply:
I would also like it to be so simple, but the problem is that we have node-trees inside node-trees so is a recursive search, and if one monad changed it has to be updated and it has to trigger updates in the upper tree and so on.
I started with something like that but solving the monad question made it more complex and returning a boolean, which is not perfect code-wise, works to inform the parent tree

Reply:
Monads have their own `update` method. Why does not put the update code there?

Reply:
The Monad is very complex node, is a node and a node-tree, it has two updates, the nodetree update and the node update, my code affects to the nodetree update which is the same as any other node-tree, thats why it is called from the nodetree class and not from the monad class

Reply:
Do you think if you put update code of `monad` tree changes into place where nobody expects to find it is making code simpler?
If you are adding new functionality to monads you should place the code into monads. Hiding functionality just make monads more complex.